checker/tests/nullness/Issue391.java

```diff
@@ -0,0 +1,50 @@
+// Test case for Issue 391:
+// https://github.com/typetools/checker-framework/issues/391
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.nullness.qual.RequiresNonNull;
+import org.checkerframework.checker.nullness.qual.EnsuresNonNull;
+
+
+class ClassA {
+    @Nullable private String value = null;
+
+    @EnsuresNonNull("value") public void ensuresNonNull() {}
+
+    @RequiresNonNull("value") public String getValue() {
+        return value;
+    }
+}
+
+public class Issue391 {
+    ClassA field = new ClassA();
+
+    @RequiresNonNull("field.value")
+    void method() {
+    }
+
+    @EnsuresNonNull("field.value")
+    void ensuresNonNull() {
+    }
+
+    void method2() {
+        ClassA a = new ClassA();
+        //:: error: (contracts.precondition.not.satisfied)
+        a.getValue();
+        //:: error: (contracts.precondition.not.satisfied)
+        method();
+    }
+
+    void method3() {
+        ensuresNonNull();
+        // TODO: the above method call should satisfy the condtion below
```

Comment:
For whom are the TODO comments? Is there a different issue for them? The test is not skipped, so should this be in a new test file that doesn't expect an unwanted error?